### DISCUSSION TITLE
Query Status

### DISCUSSION THREAD

**Discussion by ironsoul** [sqlalchemy/sqlalchemy#7838] [Usage Questions]
I am looking at running a bunch of updates in sequence. The sequence and the query for the update is stored in a table. So I would like to loop through the queries and run then based on their sequence. 

Is there a way to get back the status of the query to make sure the current query has finished before I run the next?

My current thought was to put the query that I get into sqlalchemy.text and execute it that way.

I am using core and not the ORM at the moment. Any advise would be greatly appreciated.

**Comment by zzzeek** [comment]
when you run a SQL query, it either succeeds or fails.  assuming you are blocking on the invocation of that query (which is most certainly the case here, unless you are spawning off worker threads or something like that), if you get past a query without it failing and throwing an error, it succeeded.

from the SQLAlchemy client/server point of view, if you have SQL queries inside of a table, you pretty much have to retrieve them to the client, put them in a text(), and then run them from the client.   The only way to run the queries all on the database is to write a server-side stored procedure that retrieves the queries and invokes them, but that would then not involve SQLAlchemy.

**Comment by ironsoul** [comment]
Thank you for the answer. I have tested everything and its working great.
